Unaccepted Memory UEFI ECR.md

```diff
@@ -0,0 +1,158 @@
+
+Instead of the normal feedback channels, we ask that all feedback on this ECR be given on this Github pull request, https://github.com/microsoft/mu_basecore/pull/66. The most up to date version of this ECR can be found there.<br>
+
+An example of a platform that uses this change to report unaccepted memory to an OS can be found in the following OVMF commit, https://github.com/AMDESE/ovmf/commit/793a755795a3580b7ee96d9478992767da331fb3.<br>
+
+### References 
+
+AMD APM Volume 2, Section 15.36 https://www.amd.com/system/files/TechDocs/24593.pdf <br>
+SEV Firmware Specification https://www.amd.com/system/files/TechDocs/56860.pdf <br>
+Github PR for comments https://github.com/microsoft/mu_basecore/pull/66 <br>
+
+## Detailed Description of the Change and Special Instructions 
+
+### Summary of Changes: 
+- Update Table 29 & 30 to add new type 
+- Update Section 7.2 AllocatePages Related Definitions 
+- Add entry describing unaccepted memory to Appendix R - Glossary
+
+~~Text Removed~~,  Text Added ***Text Added*** ,  New Sections ***New Section***, Discussion/Mode Edits needed ***DME***
+
+#### 7.2 Memory Allocation Services
+**Table 29. Memory Type Usage before ExitBootServices()**<br>
+| Mnemonic                | Description                                                                                                                                                                                                                                                          |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| EfiReservedMemoryType   | Not usable.                                                                                                                                                                                                                                                          |
+| ...                     | ...                                                                                                                                                                                                                                                                  |
+| EfiUnacceptedMemoryType ***Text Added*** | A memory region that represents unaccepted memory, that must be accepted by the guest before it can be used. Unless otherwise noted, all other EFI memory types are accepted. For platforms that support unaccepted memory, all unaccepted valid memory will be reported as unaccepted in the memory map. Unreported physical address ranges must be treated as not-present memory. ***Text Added*** |

```

Comment:
@jyao1 I've changed the description to describe memory that the guest should expect is usable, but unaccepted, and added a final sentence about not reported ranges. I don't think we want to require platforms to have the concept MMIO, but describing it as "not-present memory" allows the platform to do whatever it wants.

Reply:
OK. Looks good.

May I know your plan for new HOB attribute (EFI_RESOURCE_ATTRIBUTE_UNACCEPTED) and new GCD type (EfiGcdMemoryTypeUnaccepted)?

Do you need any help on that? If so, I can write some initial content then we can collaborate to improve/align the word.

Reply:
FYI I have the ECR included in the PR. Please let me know if anything is missing, thanks!